### PR TITLE
python3

### LINKED ISSUE #612: migrate to python3
I have now merged #610 which runs the tests on both python2 and python3 in CI.
The python3 build is failing at this stage. Note Github Actions does not have the concept of "allowed failure" so we're back to a red build on master :( but we should be keeping the "CKAN 2.9-py2" build green at least
For the moment we are still deploying and developing on python2
The next stage is to get to a stage where we've got a python2/3 compatible codebase (using future/six/etc) and get the test suite passing on python3. Then we can start doing some manual tests.

Currently the build is failing at the "Initialize DB" step of the build - we can't even apply the DB migrations so step 1 is to get to a stage where that passes and we can actually invoke pytest to get some specific test failures. That might unavoidably end up being a fairly big PR as bootstrapping the plugin actually loads quite a lot of code but lets scope that as tightly as possible.

Once we've done that we can start tacking test failures in manageable chunks. Ideally what I'd like to do is attack this in small chunks. A series of small PRs like "get the access request controller tests passing under python 3" is going to be much easier to review and there will be less friction to getting stuff reviewed/merged quickly.

### PR DESCRIPTION
**Pull request by chris48s**
Closes #612
Refs https://github.com/okfn/docker-ckan-unhcr-aws/pull/48

- I rolled back the commit which run the python3 tests on master in https://github.com/okfn/ckanext-unhcr/commit/d4b9d9aad31bcd757c0b9e297a936d96e84be3c0 Merging this PR will drop python2 compatibility and testing completely

- This PR gets us to the stage where all the tests pass on python3 I also did a load of manual testing and couldn't find any further issues

- There are various other bits of python2 cruft which I have chosen not to remove yet. For example, we can change all the `import mock` statements to `from unittest import mock`. We can remove all the `# -*- coding: utf-8 -*-` statements. We can also drop any leading `u`s from strings (i.e: `u'things'` --> `'things'`) etc. I have decided not to do these yet. Since we may end up "maintaining" this diff alongside master for a while I want to keep the changeset as small as possible: minimum viable python3-compatibility, if you will :) We can do this kind of "cleanup" in a follow-on PR

### CODE REVIEW

File: bin/install-extensions.bash [hunk 1]
```diff
@@ -1,22 +1,17 @@
 #!/bin/bash
 set -euo pipefail
 
+pip install setuptools-rust
```
Comment (amercader) [inline bin/install-extensions.bash:4]: Is this need to install the cryptography package?

File: ckanext/unhcr/helpers.py [hunk 2]
```diff
@@ -1028,3 +1028,8 @@ def nl_to_br(text):
     result = u'\n\n'.join(u'<p>%s</p>' % p.replace('\n', Markup('<br>\n'))
                           for p in _paragraph_re.split(escape(text)))
     return Markup(result)
+
+
+def is_plugin_loaded(plugin):
```
Comment (amercader) [inline ckanext/unhcr/helpers.py:1033]: Use this instead, it is more reliable across all scenarios:

```python
import ckan.plugins as plugins

def is_plugin_loaded(plugin_name):
     return plugins.plugin_loaded(plugin_name)
```

**Review by amercader — APPROVED** [review]
Great stuff @chris48s!

File: bin/install-extensions.bash (hunk 1 above)
Comment (chris48s) [reply bin/install-extensions.bash:4]: Yes exactly.
I've made the same change in the Dockerfiles in the deploy repo. The prod Dockerfile also needed `libffi-dev` for the cryptography install to work (the dev Dockerfile already has it)

**Comment by chris48s** [comment]
Plan for deploying this:

- Merge this PR https://github.com/okfn/ckanext-unhcr/pull/621
- Write a changelog entry
- Bump https://github.com/okfn/ckanext-unhcr/blob/master/ckanext/unhcr/__init__.py#L1
- Push a tag to this repo
- Merge https://github.com/okfn/docker-ckan-unhcr-aws/pull/48
- Bump ckanext-unhcr tag in prod `Dockerfile` e.g: https://github.com/okfn/docker-ckan-unhcr-aws/commit/b3e180f5c7ba2e6536e4ef89928b23ae0b7df1ac
- Deploy

**Pull request merged**